subprojects/docs/docs.gradle

```diff
@@ -447,17 +280,109 @@ tasks.register("checkstyleApi", Checkstyle) {
     reports.xml.destination = file("$checkstyle.reportsDir/checkstyle-api.xml")
 }
 
-def distDocs = tasks.register("distDocs", Docbook2Xhtml) {
-    dependsOn userguideAsciidoc
-    source new File(asciidocOutputDir, 'getting_started.xml')
-    destFile = new File(distDocsDir, 'getting-started.html')
-    stylesheetName = 'standaloneHtml.xsl'
+def distDocs = tasks.register("distDocs", AsciidoctorTask) {
+    sourceDir = userguideSrcDir
+    sources { include 'getting-started.adoc' }
+    outputDir = distDocsDir
+    separateOutputDirs = false
+    options doctype: 'book'
+    backends 'html5'
+}
+
+asciidoctorj {
+    noDefaultRepositories = true
+    version = '1.5.7'
 }
 
-def userguide = tasks.register("userguide") {
-    dependsOn userguideHtml, userguideSingleHtml, userguidePdf
-    description = 'Generates the userguide'
+def userguideAsciidoc = tasks.register("userguideAsciidoc", AsciidoctorTask) {
+    dependsOn css, samples, defaultImports
+    sourceDir = userguideSrcDir
+
+    inputs.file "$srcDocsDir/css/manual.css"
+    inputs.dir file("src/main/resources")
+    inputs.dir samplesSrcDir
+    sources {
+        include '*.adoc'
+        exclude 'javaProject*Layout.adoc'
+    }
+    outputDir = userguideIntermediateOutputDir
+    // Verbose mode reports broken links. See asciidoctor/asciidoctor-gradle-plugin#233
+    requires file('src/docs/ruby/verbose-mode.rb')
+    separateOutputDirs = false
+    options doctype: 'book'
+    backends = ['html5']
+
+    attributes stylesdir      : '../css/',
+        stylesheet            : 'manual.css',
```

Comment:
The new userguide has strange admonitions (single page & multi page):
![image](https://user-images.githubusercontent.com/423186/42881986-a4fb6d80-8a98-11e8-87bb-523810856ccb.png)

Before the conversion it seems that only the single page userguide had those strange symbols.

Reply:
In order to avoid pulling down some font library I made icons out of emoji + CSS trickery. The light bulb isn't super neat here. I'm trying to be resourceful without making custom icons or importing them. Do you have any suggestions?

Reply:
Consider something like [this site](https://www.flaticon.com/search?word=info) perhaps. I would use an "info" icon for `NOTE`, a lightbulb for `TIP`, and some form of exclamation point for `CAUTION` and/or `WARNING` (I'm not sure if they are both admonitions in Asciidoctor or not).

Reply:
And I would use SVG for the icons if possible.

Reply:
Okay, I'd like to do more style polishing in a separate PR after getting these base changes in. For now I have changed the `INFO` admonition. `CAUTION` and `WARNING` are admonitions as well. 

Note that I'd like to merge the CSS and JS for guides and docs, and possibly just host on assets.gradle.com. This would mean we should make changes then, as there are quite a few little differences to work out.

Reply:
That's fine with me. I am also fine with having no icons at all as we had before.